swap/protocol.go

```diff
@@ -134,17 +136,22 @@ func (s *Swap) removePeer(p *Peer) {
 	delete(s.peers, p.ID())
 }
 
-func (s *Swap) addPeer(p *Peer) {
+func (s *Swap) addPeer(protoPeer *protocols.Peer, beneficiary common.Address, contractAddress common.Address) (*Peer, error) {
 	s.peersLock.Lock()
 	defer s.peersLock.Unlock()
+	p, err := NewPeer(protoPeer, s, beneficiary, contractAddress)
+	if err != nil {
+		return nil, err
+	}
 	s.peers[p.ID()] = p
+	return p, nil
 }
 
-func (s *Swap) getPeer(id enode.ID) (*Peer, bool) {
+func (s *Swap) getPeer(id enode.ID) *Peer {
 	s.peersLock.RLock()
 	defer s.peersLock.RUnlock()
-	peer, ok := s.peers[id]
-	return peer, ok
+	peer := s.peers[id]
```

Comment:
why are we removing the `ok` check here? What if the peer is not in the map? Do you just want to force the caller to check for `nil`?

Reply:
Yes, this was suggested here (https://github.com/ethersphere/swarm/pull/1725#discussion_r320438711)